### PR TITLE
[doc] Convert custom datetime column when reading a CSV file

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

Give an example of how a datetime column can be parsed when reading a CSV file.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by c21 — APPROVED** [review]
LGTM

File: python/ray/data/read_api.py [hunk 1]
```diff
@@ -562,6 +562,14 @@ def read_csv(
         >>> ray.data.read_csv( # doctest: +SKIP
         ...     ["s3://bucket/path1", "s3://bucket/path2"])
 
+        >>> # Convert a date column with a custom format from a CSV file.
+        >>> from pyarrow import csv
+        >>> convert_options = csv.ConvertOptions(
+        ...     timestamp_parsers=["%m/%d/%Y"])
```
Comment (matthewdeng) [inline python/ray/data/read_api.py:568]: Would it make sense to generalize this example to all `convert_options` ([pyarrow.csv.ConvertOptions](https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html)) or even `arrow_csv_args`?

In other words, should we delegate this documentation over to Arrow?

Comment (pcmoritz) [reply python/ray/data/read_api.py:568]: Yes, part of my desire to document this is to make sure that people know how to put in arbitrary `ConvertOptions` (and I wanted to show one that is probably particularly useful as an example). I'll add a note about the more general case and also link the Arrow documentation :)

**Comment by DmitriGekhtman** [comment]
oops, giant diff

**Comment by pcmoritz** [comment]
Haha yeah, I fixed it now :)

**Pull request merged**